pandas/tests/io/test_excel.py

```diff
@@ -1058,351 +1008,291 @@ def test_read_excel_squeeze(self):
-            self.frame.to_excel(path, 'test1', header=False)
-            self.frame.to_excel(path, 'test1', index=False)
-
-    def test_int_types(self):
-        _skip_if_no_xlrd()
-
-        for np_type in (np.int8, np.int16, np.int32, np.int64):
-
-            with ensure_clean(self.ext) as path:
-                # Test np.int values read come back as int (rather than float
-                # which is Excel's format).
-                frame = DataFrame(np.random.randint(-10, 10, size=(10, 2)),
-                                  dtype=np_type)
-                frame.to_excel(path, 'test1')
-                reader = ExcelFile(path)
-                recons = read_excel(reader, 'test1')
-                int_frame = frame.astype(np.int64)
-                tm.assert_frame_equal(int_frame, recons)
-                recons2 = read_excel(path, 'test1')
-                tm.assert_frame_equal(int_frame, recons2)
-
-                # test with convert_float=False comes back as float
-                float_frame = frame.astype(float)
-                recons = read_excel(path, 'test1', convert_float=False)
-                tm.assert_frame_equal(recons, float_frame,
-                                      check_index_type=False,
-                                      check_column_type=False)
-
-    def test_float_types(self):
-        _skip_if_no_xlrd()
-
-        for np_type in (np.float16, np.float32, np.float64):
-            with ensure_clean(self.ext) as path:
-                # Test np.float values read come back as float.
-                frame = DataFrame(np.random.random_sample(10), dtype=np_type)
-                frame.to_excel(path, 'test1')
-                reader = ExcelFile(path)
-                recons = read_excel(reader, 'test1').astype(np_type)
-                tm.assert_frame_equal(frame, recons, check_dtype=False)
-
-    def test_bool_types(self):
-        _skip_if_no_xlrd()
-
-        for np_type in (np.bool8, np.bool_):
-            with ensure_clean(self.ext) as path:
-                # Test np.bool values read come back as float.
-                frame = (DataFrame([1, 0, True, False], dtype=np_type))
-                frame.to_excel(path, 'test1')
-                reader = ExcelFile(path)
-                recons = read_excel(reader, 'test1').astype(np_type)
-                tm.assert_frame_equal(frame, recons)
-
-    def test_inf_roundtrip(self):
-        _skip_if_no_xlrd()
-
+        df.to_excel(self.path, 'test1')
+        reader = ExcelFile(self.path)
+        recons = read_excel(reader, 'test1')
+        tm.assert_frame_equal(df, recons)
+
+    def test_basics_with_nan(self, merge_cells, engine, ext):
+        self.frame['A'][:5] = nan
+        self.frame.to_excel(self.path, 'test1')
+        self.frame.to_excel(self.path, 'test1', columns=['A', 'B'])
+        self.frame.to_excel(self.path, 'test1', header=False)
+        self.frame.to_excel(self.path, 'test1', index=False)
+
+    @pytest.mark.parametrize("np_type", [
```

Comment:
Figured I might as well parametrize this and the following two functions as part of this